### ISSUE TITLE
Debug formatting for pointers is truncated

### ISSUE THREAD

**Issue by SSheldon** [rust-lang/rust#22854]
``` rust
fn main() {
	let x = 0;
	let p = &x as *const i32;
	println!("{:?}, {:x}", p, p as usize);
}
```

This sample code prints `0x5a83172c, 7fff5a83172c`, where the pointer's debug doesn't include the first 4 characters of its address.

My rustc version:
```
$ rustc --version --verbose
rustc 1.0.0-nightly (4db0b3246 2015-02-25) (built 2015-02-25)
binary: rustc
commit-hash: 4db0b32467535d718d6474de7ae8d1007d900818
commit-date: 2015-02-25
build-date: 2015-02-25
host: x86_64-apple-darwin
release: 1.0.0-nightly
```

**Comment by SSheldon** [comment]
Possibly related to #22700, the most recent change to the `fmt` module? cc @nrc

**Comment by SSheldon** [comment]
Fixed in #22901
